### ISSUE TITLE
Adding Zero123 and other LDM enabled 3D reconstruction models

### ISSUE THREAD

**Issue by pgn-dev** [huggingface/diffusers#4096]
### Model/Pipeline/Scheduler description

Recent advances in multiview synthesis with 2D LDMs has been made possible by training LDMs on camera extrinsics. This research has enabled a high-fidelity 3D reconstruction with a single input image using implicit models (NeRFs, SDFs).

While there are several papers that use vanilla LDMs (those not conditioned on camera extrinsics) for 3D reconstruction via score distillation, models like Zero123 that are conditioned on camera extrinsics show greater promise in terms of fidelity.

I propose to extend diffusers by adding two new pipelines:

a) A multiview synthesis pipeline using LDMs conditioned on an image and camera spherical coordinates (radius, azimuth, elevation).
b) A 3D reconstruction pipeline capable of creating mesh and texture using the synthesis pipeline from point a.

These pipelines, once implemented, could significantly improve the fidelity of 3D reconstructions. It could also provide a foundation for future projects in 3D modeling and reconstruction.### 

### Open source status

- [X] The model implementation is available
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Official Zero123 implementation: https://github.com/cvlab-columbia/zero123 @ruoshiliu

Zero123-XL trained on Objaverse-XL: https://objaverse.allenai.org/objaverse-xl-paper.pdf

Fairly comprehensive project which covers major 3D reconstruction papers along with a link to Zero123-XL checkpoints: https://github.com/threestudio-project/threestudio @voletiv @bennyguo

**Comment by kxhit** [comment]
Hi @pgn-dev @ruoshiliu I just implemented a diffuser pipeline for zero123 at [here](https://github.com/kxhit/zero123-hf). I feel diffusers pipeline/codebase is much easier and cleaner than the original pytorch lightning implementation.

Thinking to merge into the main diffuser repo and the training script is working in progress.  

Any suggestions or contributions are welcomed!

**Comment by bennyguo** [comment]
Hi @pgn-dev @kxhit @ruoshiliui I also opened a PR about this: https://github.com/huggingface/diffusers/pull/4417. I've converted the original Zero123 checkpoint to the diffusers format and uploaded the converted weights at https://huggingface.co/bennyguo/zero123-diffusers.

**Comment by bennyguo** [comment]
@voletiv I did: https://github.com/huggingface/diffusers/blob/bc691231360a4cbc7d19a58742ebb8ed0f05e027/scripts/convert_zero123_to_diffusers.py

**Comment by sayakpaul** [comment]
Completely missed out on this! I think starting with a community pipeline is a great idea and then upon community interest, we can graduate them to become core pipelines. 

We truly appreciate the discussions here!

**Comment by sayakpaul** [comment]
Seems like we have two PRs for the same pipeline?

https://github.com/huggingface/diffusers/pull/4295

https://github.com/huggingface/diffusers/pull/4417

@kxhit and @bennyguo would you two maybe like to collaborate on a single PR and take things from there? I think that will be easier. 

https://github.com/huggingface/diffusers/pull/4417 has got a lot of merge conflicts and the commits are probably broken as they show commits irrelevant to the PR. So, I would be in favor of closing them.

Also ccing @yiyixuxu since she recently contributed the support for the first-ever 3D pipeline in diffusers - SHAP-E.

**Comment by bennyguo** [comment]
The conflicts are due to that I start from a previous tag instead of `main`. Only two files are added without modifying any existing files. I think https://github.com/huggingface/diffusers/pull/4295 still relies on the original checkpoint while I provide a conversion script to convert the original checkpoint to diffusers format.

**Comment by bennyguo** [comment]
Shall we make this a built-in pipeline?

**Comment by sayakpaul** [comment]
I would then be in favor of a new PR as the number of conflicts is way too high to resolve and probably isn't worth it. A new PR is easier IMO.

**Comment by bennyguo** [comment]
Sure! Will open a new PR today. I also have to fix the "extra module" problem we discussed on Slack 😂

**Comment by kxhit** [comment]
Thanks a lot for reviewing. The https://github.com/huggingface/diffusers/pull/4295 PR converted standard model ckpts to diffusers e.g.unet, vae, image_encoder. The cc_projection is not yet converted but I think it is easy to do so. I will try to update and thanks @bennyguo for sharing the convert scripts.

**Comment by bennyguo** [comment]
@kxhit Great! Are you still using the original checkpoint to load the cc_projection layer or did you make the layer as a CLIPCameraProjection module as in my setting?

**Comment by kxhit** [comment]
@bennyguo All weights in diffusers now! No more original ckpt required.
